src/lib/__tests__/routing/spa-replace-state-test.ts

```diff
@@ -0,0 +1,82 @@
+import { createSearchClient } from '../../../../test/mock/createSearchClient';
+import { wait } from '../../../../test/utils/wait';
+import historyRouter from '../../routers/history';
+import instantsearch from '../../..';
+import { connectSearchBox } from '../../../connectors';
+
+const writeDelay = 10;
+const writeWait = 1.5 * writeDelay;
+
+describe('routing', () => {
+  beforeEach(() => {
+    window.history.replaceState({}, '', 'http://localhost/');
+    jest.clearAllMocks();
+  });
+
+  // It behaves like this because we check the `window.history.length`
+  // to prevent `write` after `dispose`
+  test('SPA (Single Page App) `replaceState` use case: URL should not be cleaned', async () => {
```

Comment:
```suggestion
  // It behaves like this because we check the `window.history.length`
  // to prevent `write` after `dispose`
  test('SPA (Single Page App) `replaceState` use case: URL is cleaned', async () => {
```

Reply:
```suggestion
  // We can't assert whether another router did update the URL
  // So there's no way to prevent `write` after `dispose`
  test('cleans the URL after navigating', async () => {
```